docs/whats_new/v0.4.0.md

```diff
@@ -0,0 +1,152 @@
+
+@Innixma, @yinweisu, @gradientsky, @zhiqiangdon, @jwmueller, @willsmithorg, @sxjscience, @DolanTheMFWizard, @truebluejason, @taesup-aws, @Xilorole, @mseeger, @killerSwitch, @rschmucker
+
+# Changes
+
+## General
+
+- AutoGluon now supports Windows OS! Both CPU and GPU are supported on Windows. @Innixma, @yinweisu (#1463, #1470)
+- AutoGluon now supports Python 3.9! @yinweisu (#1330)
+- Python 3.6 support has been dropped.
+- AutoGluon no longer depends on MXNet. PyTorch is now used for all deep learning models.
+- AutoGluon no longer depends on ConfigSpace, cython, dill, paramiko, autograd, openml, d8, graphviz and many more packages.
+This greatly simplifies installation of AutoGluon, particularly on Windows. @Innixma
+- Entirely refactored HPO logic to break dependencies on ConfigSpace and improve stability and ease of development.
+HPO has been simplified to use random search in this release while we work on
+re-introducing the more advanced HPO methods such as bayesopt in a future release.
+Additionally, removed 40,000 lines of out-dated code to streamline future development.
+@Innixma (#1397, #1411, #1414, #1431, #1443, #1511)
+- Added cloud training with AWS SageMaker tutorials. @gradientsky (#1375)
+- Added GPU training tutorial for tabular models. @gradientsky (#1527)
+- Added feature preprocessing tutorial for tabular. @willsmithorg (#1478)
+- Added `autogluon.common` to simplify dependency management for future submodules. @Innixma (#1386)
+- Removed `autogluon.mxnet` and `autogluon.extra` submodules as part of code cleanup. @Innixma (#1397, #1411, #1414)
+- Refactored logging to avoid interfering with other packages. @yinweisu (#1403)
+- Fixed logging output on Kaggle, previously no logs would be displayed while fitting AutoGluon in a Kaggle kernel. @Innixma (#1468)
+- Added platform tests for Linux, MacOS, and Windows. @yinweisu (#1464, #1506, #1513)
+- Added ROADMAP.md to highlight past, present, and future feature prioritization and progress to the community. @Innixma (#1420)
+- Various documentation and CI improvements
+  - @jwmueller (#1379, #1408, #1429)
+  - @gradientsky (#1383, #1387, #1471, #1500)
+  - @yinweisu (#1441, #1482, #1566, #1580)
+  - @willsmithorg (#1476, #1483)
+  - @Xilorole (#1526)
+  - @Innixma (#1452, #1453, #1528, #1577, #1584, #1588, #1593)
+- Various backend enhancements / refactoring / cleanup
+  - @DolanTheMFWizard (#1319)
+  - @gradientsky (#1320, #1366, #1385, #1448, #1488, #1490, #1570, #1576)
+  - @mseeger (#1349)
+  - @yinweisu (#1497, #1503, #1512, #1563, #1573)
+  - @willsmithorg (#1525, #1543)
+  - @Innixma (#1311, #1313, #1327, #1331, #1338, #1345, #1369, #1377, #1380, #1408, #1410, #1412, #1419, #1425, #1428, #1462, #1465, #1562, #1569, #1591, #1593)
+- Various bug fixes
+  - @jwmueller (#1314, #1356)
+  - @yinweisu (#1472, #1499, #1504, #1508, #1516)
+  - @gradientsky (#1514)
+  - @Innixma (#1304, #1325, #1326, #1337, #1365, #1395, #1405, #1587, #1599)
+
+## Text
+
+TODO
```

Comment:
Tentative release note about text. (Not sure if we should separately mention AutoMMPredictor)


In 0.4, we made a major refactor of AutoGluon Text. Now, AutoGluon Text is implemented with [PyTorch Lightning](https://www.pytorchlightning.ai/), and supports backbones available in [huggingface/transformers](https://huggingface.co/docs/transformers/index). Compared with the TextPredictor in AutoGluon 0.3, the TextPredictor in AutoGluon 0.4 has 72.22% win-rate in the multimodal text-tabular benchmark published in NeurIPS 2021 (https://arxiv.org/abs/2111.02705). If we use the “high_quality” preset, the win-rate increased to 77.8% thanks to the DeBERTa-v3 backbone (https://arxiv.org/abs/2111.09543). The new version has ~2.88x speedup in training and ~1.40x speedup in inference. With g4dn.12x instance, the model can further achieve 2.26x speedup with 4 GPUs.
In addition, we resubmitted our results to “MachineHack: Product Sentiment Analysis”, “MachineHack: Predict the Price of Books”, “Kaggle: Mercari Price Suggestion”. With three lines of code, AutoGluon 0.4 is able to achieve top places in these competitions (1st, 2nd, 2nd correspondingly). In addition, the results obtained by AutoGluon 0.4 consistently outperforms the results obtained by AutoGluon 0.3 for these three competitions.
Moreover, we added two new features:
- Support solving multilingual problems and cross-lingual transfer: https://auto.gluon.ai/dev/tutorials/text_prediction/multilingual_text.html#cross-lingual-transfer. You can now train a model on English dataset and directly apply the model on datasets in other languages such as German, Japanese, Italian, …
- An experimental AutoMMPredictor that supports flexible fusion of image backbones (those in TIMM), text backbone (those in huggingface/transformers), and multimodal backbones like CLIP.

Reply:
Could maybe add one more sentence following "An experimental AutoMMPredictor..." which explains:  

The AutoMMPredictor is a single large neural network for supervised learning with both image and text data.  When there exist complex relationships between the modalities, AutoMMPredictor may perform better than the ImagePredictor+TextPredictor ensemble demonstrated in this tutorial: https://auto.gluon.ai/stable/tutorials/tabular_prediction/tabular-multimodal.html

Reply:
## Text

AutoGluon Text is refactored with [PyTorch Lightning](https://www.pytorchlightning.ai/). It now supports backbones in [huggingface/transformers](https://huggingface.co/docs/transformers/index). The new version has better performance and is also faster. In addition, support solving multilingual problems and a new `AutoMMPredictor` for automatically building multimodal DL models.


- **Better Performance**
   - Compared with TextPredictor in AutoGluon 0.3, TextPredictor in AutoGluon 0.4 has **72.22%** win-rate in the [multimodal text-tabular benchmark pubished in NeurIPS 2021](https://arxiv.org/abs/2111.02705). If we use `presets="high_quality"`, the win-rate increased to **77.8%** thanks to the [DeBERTa-v3 backbone](https://arxiv.org/abs/2111.09543).
   - In addition, we resubmitted our results to [MachineHack: Product Sentiment Analysis](https://machinehack.com/hackathon/product_sentiment_classification_weekend_hackathon_19/overview
), ["MachineHack: Predict the Price of Books"](https://machinehack.com/hackathon/predict_the_price_of_books/overview
), ["Kaggle: Mercari Price Suggestion"](https://www.kaggle.com/c/mercari-price-suggestion-challenge). With three lines of code, AutoGluon 0.4 is able to achieve top places in these competitions (1st, 2nd, 2nd correspondingly). The results obtained by AutoGluon 0.4 also consistently outperform the results obtained by AutoGluon 0.3.
- **Faster Speed**
   - The new version has ~2.88x speedup in training and ~1.40x speedup in inference. With g4dn.12x instance, the model can further achieve 2.26x speedup with 4 GPUs.
- **Multilingual Support**
   - AutoGluon Text now supports solving multilingual problems via cross-lingual transfer. This is triggered by setting `presets="multilingual"`. You can now train a model on the English dataset and directly apply the model on datasets in other languages such as German, Japanese, Italian, etc.
- **AutoMMPredictor for Multimodal Problems**
   - Support an experimental AutoMMPredictor that supports fusion image backbones in [timm](https://github.com/rwightman/pytorch-image-models/tree/master/timm), text backbone in [huggingface/transformers](https://huggingface.co/docs/transformers/index), and multimodal backbones like [CLIP](https://openai.com/blog/clip/). It may perform better than ensembling ImagePredictor + TextPredictor.

Reply:
@Innixma You may use the version above.

Reply:
## Text

AutoGluon Text is refactored with [PyTorch Lightning](https://www.pytorchlightning.ai/). It now supports backbones in [huggingface/transformers](https://huggingface.co/docs/transformers/index). The new version has better performance and is also faster. In addition, we now support solving multilingual problems and added a new `AutoMMPredictor` for automatically building multimodal DL models.


- **Better Performance**
   - Compared with TextPredictor in AutoGluon 0.3, TextPredictor in AutoGluon 0.4 has **72.22%** win-rate in the [multimodal text-tabular benchmark pubished in NeurIPS 2021](https://arxiv.org/abs/2111.02705). If we use `presets="high_quality"`, the win-rate increased to **77.8%** thanks to the [DeBERTa-v3 backbone](https://arxiv.org/abs/2111.09543).
   - In addition, we resubmitted our results to [MachineHack: Product Sentiment Analysis](https://machinehack.com/hackathon/product_sentiment_classification_weekend_hackathon_19/overview
), ["MachineHack: Predict the Price of Books"](https://machinehack.com/hackathon/predict_the_price_of_books/overview
), ["Kaggle: Mercari Price Suggestion"](https://www.kaggle.com/c/mercari-price-suggestion-challenge). With three lines of code, AutoGluon 0.4 is able to achieve top places in these competitions (1st, 2nd, 2nd correspondingly). The results obtained by AutoGluon 0.4 also consistently outperform the results obtained by AutoGluon 0.3.
- **Faster Speed**
   - The new version has **~2.88x** speedup in training and **~1.40x** speedup in inference. With g4dn.12x instance, the model can further achieve 2.26x speedup with 4 GPUs.
- **Multilingual Support**
   - AutoGluon Text now supports solving multilingual problems via cross-lingual transfer. This is triggered by setting `presets="multilingual"`. You can now train a model on the English dataset and directly apply the model on datasets in other languages such as German, Japanese, Italian, etc.
- **AutoMMPredictor for Multimodal Problems**
   - Support an experimental AutoMMPredictor that supports fusion image backbones in [timm](https://github.com/rwightman/pytorch-image-models/tree/master/timm), text backbone in [huggingface/transformers](https://huggingface.co/docs/transformers/index), and multimodal backbones like [CLIP](https://openai.com/blog/clip/). It may perform better than ensembling ImagePredictor + TextPredictor.
- **Other Features**
   - Support continuous training from an existing checkpoint. You may just call `.fit()` again after a previous trained model has been loaded.

Reply:
Thanks, added!